File: frontend/miniprogram/src/api/reading-expression.ts
import { request } from '@/utils/request'

export interface REGradePoint { point: string; hit: boolean; comment?: string }

export interface REGradeResult {
  points: REGradePoint[]
  content_score: number
  content_full: number
  language_comment: string
  language_deduction: number
  total: number
  full: number
  feedback: string
}

/** 阅读表达简答 AI 批改(P2a):逐要点命中 + 内容/语言得分 + 反馈。 */
export function gradeReadingExpression(payload: {
  question: string
  reference_answer: string
  student_answer: string
  passage?: string
  full_score?: number
}): Promise<REGradeResult> {
  return request<REGradeResult>('/api/v1/reading-expression/grade', {
    method: 'POST',
    data: payload,
  })
}

File: backend/app/schemas/reading_expression.py
"""阅读表达批改 schema（P2a）。"""
from __future__ import annotations

from pydantic import BaseModel, Field


class ReadingExpressionGradeIn(BaseModel):
    """一道阅读表达简答的批改请求。"""
    question: str = Field(..., min_length=1, description="题目/问题")
    reference_answer: str = Field(..., min_length=1, description="参考答案")
    student_answer: str = Field("", description="学生作答(空则返回未作答)")
    passage: str | None = Field(None, description="所属短文(可选,判分参考)")
    full_score: int = Field(4, ge=1, le=20, description="本题满分")
